pkg/util/iptables/save_restore.go

```diff
@@ -55,3 +56,40 @@ func GetChainsFromTable(save []byte) sets.Set[Chain] {
 	}
 	return chainsSet
 }
+
+// GetCountersForRule parse and returns packet and byte counters associated with the rule.
+func GetCountersForRule(save []byte, rule string) (int, int, error) {
+	// \[(\d+):(\d+)] -A KUBE-FORWARD -m conntrack --ctstate INVALID -j DROP
+	var err error
+	index := bytes.Index(save, []byte(rule))
+	pCounter, bCounter := 0, 0
+	if index == -1 {
+		return pCounter, bCounter, fmt.Errorf("rule not found")
+	}
+
+	start, end := index, index
+	exit := false
+	for start >= 0 && !exit {
```

Comment:
This seems like a weird way to parse this.
I probably would use just used a regexp. If not, I would have checked that `save[0] == '['` and then found `strings.Index(save, ":")` and `strings.Index(save, "]")`, and if they exist and are in the right order, then try to parse the numbers between them.

Reply:
I initially thought about regex only (the pattern is still in the code, commented out) but wanted it to be as fast as possible so I ended up with leet-code style. I will do a benchmark to compare the approaches.

Reply:
You want to aim for "fast but not leet"; future us has to understand this code too, and making `GetCountersForRule` twice as fast isn't going to help much anyway when `iptables-restore` is so slow. (Amdahl's Law)

Reply:
updated with regex match.